include/zephyr/arch/cache.h

```diff
@@ -121,6 +127,9 @@ extern int arch_dcache_invd_range(void *addr, size_t size);
  *
  * Flush and Invalidate the specified address range of the data cache.
  *
+ * @note when @p addr is not aligned to the cache line and/or @p size is not
```

Comment:
I would say that invalidating data structures sharing cache lines with other data structures, which can be written by any software module can lead to data corruption and the behavior is undefined.

If an invalidated data structure shares cache lines with read-only data, then it's fine.